app/models/krikri/provider.rb

```diff
@@ -69,7 +47,8 @@ def self.all
 
       response.facet_pivot['provider_id,provider_name'].map do |facet|
         rdf_subject = facet['value']
-        name = facet['pivot'].present? ? facet['pivot'].first['value'] : nil
+        name = facet['pivot'].present? ? facet['pivot'].first['value'] :
+          rdf_subject
```

Comment:
This corrects an oversight in PR #180 and ensures that in all cases in which the `name` attribute is initialized, it will return the provider's full URI if the `name` value is nil.